PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -660,8 +658,15 @@ function Check-ExtractZip(
 
         Show-Update "Extracting $Path -> $ExtractToPath"
 
-        try { [System.IO.Compression.ZipFile]::ExtractToDirectory($Path, $ExtractToPath) }
-        catch { Show-Error("Can't extract results as Zip file from '$Path' to '$ExtractToPath'") }
+        try 
+        { 
+            Add-Type -Assembly System.IO.Compression.FileSystem

```

Comment:
We might just need to do this Add-Type just once instead of in all places?

Reply:
After removing script block instantiation, probably we don't need this everywhere.
Sure, let me try that.

Reply:
It cannot be placed in common block, as initblock is executed in untrusted execution.
I have called add-type everywhere before using compression command, anything specific you have in mind?